engine/apps/user_management/utils.py

```diff
@@ -0,0 +1,35 @@
+import typing
+
+from apps.user_management.constants import (
+    AlertGroupTableColumn,
+    AlertGroupTableColumns,
+    AlertGroupTableColumnTypeChoices,
+    AlertGroupTableDefaultColumnChoices,
+)
+
+if typing.TYPE_CHECKING:
+    from apps.user_management.models import User
+
+
+def default_columns() -> typing.List[AlertGroupTableColumn]:
+    columns = [
+        {"name": column.label, "id": column.value, "type": AlertGroupTableColumnTypeChoices.DEFAULT.value}
+        for column in AlertGroupTableDefaultColumnChoices
+    ]
+    return columns
+
+
+def alert_group_table_user_settings(user: "User") -> AlertGroupTableColumns:
+    organization_columns = user.organization.alert_group_table_columns
+    visible_columns: typing.List[AlertGroupTableColumn]
+    if user.alert_groups_table_selected_columns:
+        visible_columns = [
+            column for column in user.alert_groups_table_selected_columns if column in organization_columns
+        ]
+    else:
+        visible_columns = default_columns()
+    user.update_alert_group_table_columns_settings(visible_columns)
```

Comment:
question: why it's necessary to make an update here? is it possible to avoid DB writes for this function?

Reply:
it's used for lazy update user settings. Another option - update it for all users at the background on changing organization settings